### ISSUE TITLE
Initialised EWM

### ISSUE THREAD

**Issue by naught101** [pydata/pandas#13638]
Currently, with an exponential weighted moving window, there are two options, adjusted, or non-adjusted. The first one results in excessively high variance at the beginning of the series, the second results in an artificially low mean. Another alternative is to intialise the series with a scalar value - most obviously the mean. Here is a comparison of how these three variations look:

![ewm_variations](https://cloud.githubusercontent.com/assets/167164/16793921/ee7ad1ae-4917-11e6-800e-5969eeab1507.png)

This is from a hacky iterative implementation I just wrote as a proof of concept, but it shouldn't be hard to write an implementation that's just as efficient as the adjusting version. If this seems useful, I'm happy to make  a PR.

What would be the best API design? Perhaps just add an `initialise=None` that accepts floats, and overrides the `adjust=` argument if it is set?

**Comment by naught101** [comment]
Yeah, it's just the mean of the whole series, and then `y[t] = alpha * x[t] + (1 - alpha) * y[t - 1]`, but I'd suggest allowing the user to specify the value manually as well.

Yeah, i'm interested in this because I want a complete series..

On 14 July 2016 12:41:49 AM AEST, Maximilian Roos <notifications@github.com> wrote:
>Do you have code / pseudo code? Where do you get the mean from? Is it
>the mean of the whole series?
>
>Generally a `min_periods` > 0 will solve some of these problems,
>although I could imagine a situation where it's OK to have forward
>information and you want to start the series immediately.
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/pydata/pandas/issues/13638#issuecomment-232376723

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by MaximilianR** [comment]
OK, makes sense.

Could you preface the series with that value and use `adjust=False`? Does that give you the same result?

**Comment by naught101** [comment]
@MaximilianR: yeah, that does work, prefix, ewm, then remove the prefix.

**Comment by jreback** [comment]
so maybe this warrants an example in the docs? (but no code change)

**Comment by naught101** [comment]
Yeah, maybe, although I'm not sure of the best way to do it efficiently this way - it requires the creation of a new DataFrame.

Also, I think using the long-term mean is a pretty sane default behaviour - it's still data-driven (assuming some level of stationarity), and it produces a much more realistic result than either of the current options.

**Comment by chris-b1** [comment]
Related [SO answer](http://stackoverflow.com/questions/37924377/does-pandas-calculate-ewm-wrong)

I think an `initialize=` keyword would be a reasonable addition, if just for convenience, though I definitely would not use the long term mean as a default.  It may make sense in some cases, but using future data in the estimate is pretty antithetical to a "moving average", in general.
